src/components/Panels/PropertiesPanel/PanelTypes/PanelType3.vue

```diff
@@ -0,0 +1,122 @@
+    />
+    <p v-if="!obj.disableLabel">
+        <span>Label:</span>
+        <input
+            class="objectPropertyAttribute"
+            type="text"
+            name="setLabel"
+            autocomplete="off"
+            :value="escapeHtml(obj.label)"
+        />
+    </p>
+    <DropdownSelect
+        v-if="!obj.labelDirectionFixed"
+        :dropdown-array="labelDirections"
+        property-name="newLabelDirection"
+        :property-value="obj.labelDirection"
+        property-input-name="Label Direction:"
+        property-select-id="labelDirectionValue"
+    />
+    <DropdownSelect
+        v-if="!obj.directionFixed"
+        :dropdown-array="labelDirections"
+        property-name="newDirection"
+        :property-value="obj.direction"
+        property-input-name="Direction:"
+        property-select-id="directionValue"
+    />
+    <DropdownSelect
+        v-if="!obj.orientationFixed"
+        :dropdown-array="labelDirections"
+        property-name="newDirection"
+        :property-value="obj.direction"
+        property-input-name="Orientation:"
+        property-select-id="orientationValue"
+    />
+
+    <div v-for="(value, name) in obj.mutableProperties" :key="name">
+        <InputGroups
+            v-if="value.type === 'number'"
+            :property-name="value.name"
+            :property-value="obj[name]"
+            property-value-type="number"
+            :value-min="value.min || '0'"
+            :value-max="value.max || '200'"
+            :property-input-name="value.func"
+            :property-input-id="value.name"
+        />
+        <p v-if="value.type === 'text'">
+            <span>{{ value.name }}:</span>
+            <input
+                class="objectPropertyAttribute"
+                type="text"
+                :name="value.func"
+                autocomplete="off"
+                :maxlength="value.maxlength || '200'"
+                :value="obj[name]"
+            />
+        </p>
+        <p v-if="value.type === 'button'" class="btn-parent">
+            <button
+                class="objectPropertyAttribute btn custom-btn--secondary"
+                type="button"
+                :name="value.func"
+            >
+                {{ value.name }}
+            </button>
+        </p>
+        <p v-if="value.type === 'textarea'">
+            <span>{{ value.name }}</span>
+            <textarea
+                class="objectPropertyAttribute"
+                type="text"
+                autocomplete="off"
+                rows="9"
+                :name="value.func"
+            >
+                {{ obj[name] }}
+            </textarea>
+        </p>
+    </div>
+</template>
+
+<script lang="ts" setup>
+import { escapeHtml } from '#/simulator/src/utils'
+import InputGroups from '../../Shared/InputGroups.vue'
+import DropdownSelect from '../../Shared/DropdownSelect.vue'
+const props = defineProps({
+    data: { type: Object, default: undefined },
+})
+const obj = props.data
```

Comment:
1/ lets use proper variable names here,
2/ storing props item in a variable like this removes the reactivity, are we intentionally removing reactivity?
3/ you can do something like `const { obj } = toRefs(props)` to store props in a variable and retain reactivity.
4/ also, the template can directly access all props, we don't need to extract it in the setup for the template to access props